binding/flutter/ios/Classes/SwiftPorcupinePlugin.swift

```diff
@@ -11,9 +11,150 @@
+                result(errorToFlutterError(error))
+            } catch {
+                result(errorToFlutterError(PorcupineError.PorcupineInternalError(error.localizedDescription)))
+            }
+            break
+        case .DELETE:
+            let args = call.arguments as! [String: Any]
+            
+            if let handle = args["handle"] as? String {
+                if let porcupine = porcupinePool.removeValue(forKey: handle) {
+                    porcupine.delete()
+                }
+            }
+            break
+        default:
+            result(errorToFlutterError(PorcupineError.PorcupineRuntimeError("Porcupine method '\(call.method)' is not a valid function.")))
+        }
+    }
+    
+    private func errorToFlutterError(_ error: PorcupineError) -> FlutterError {
+        switch(error) {
+        case .PorcupineOutOfMemoryError:
+            return FlutterError(code: "PorcupineMemoryException", message: extractMessage("\(error)"), details: nil)
+        case .PorcupineIOError:
+            return FlutterError(code: "PorcupineIOException", message: extractMessage("\(error)"), details: nil)
+        case .PorcupineInvalidArgumentError:
+            return FlutterError(code: "PorcupineInvalidArgumentException", message: extractMessage("\(error)"), details: nil)
+        case .PorcupineStopIterationError:
+            return FlutterError(code: "PorcupineStopIterationException", message: extractMessage("\(error)"), details: nil)
+        case .PorcupineKeyError:
+            return FlutterError(code: "PorcupineKeyException", message: extractMessage("\(error)"), details: nil)
+        case .PorcupineInvalidStateError:
+            return FlutterError(code: "PorcupineInvalidStateException", message: extractMessage("\(error)"), details: nil)
+        case .PorcupineRuntimeError:
+            return FlutterError(code: "PorcupineRuntimeException", message: extractMessage("\(error)"), details: nil)
+        case .PorcupineActivationError:
+            return FlutterError(code: "PorcupineActivationException", message: extractMessage("\(error)"), details: nil)
+        case .PorcupineActivationLimitError:
+            return FlutterError(code: "PorcupineActivationLimitException", message: extractMessage("\(error)"), details: nil)
+        case .PorcupineActivationThrottledError:
+            return FlutterError(code: "PorcupineActivationThrottledException", message: extractMessage("\(error)"), details: nil)
+        case .PorcupineActivationRefusedError:
+            return FlutterError(code: "PorcupineActivationRefusedException", message: extractMessage("\(error)"), details: nil)
+        case .PorcupineInternalError:
+            return FlutterError(code: "PorcupineException", message: extractMessage("\(error)"), details: nil)
+        }
+    }
     
-  }
+    private func extractMessage(_ errorMessage: String) -> String {
+        let parts = errorMessage.components(separatedBy: "\"")
```

Comment:
Is there not a way to just get the message from the error type? This string parsing seems a bit unnecessary.